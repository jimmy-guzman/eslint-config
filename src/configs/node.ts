import nodePlugin from "eslint-plugin-n";

import type { TypedConfigItem } from "../types";

import { nodeRules } from "../rules/node";

export const nodeConfig = () => {
  return [
    {
      name: "jimmy.codes/node",
      plugins: {
        n: nodePlugin,
      },
      rules: nodeRules,
    },
  ] satisfies TypedConfigItem[];
};
