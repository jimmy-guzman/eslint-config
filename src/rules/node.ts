import type { Rules } from "../types";

export const nodeRules = {
  "n/prefer-node-protocol": "error",
} satisfies Rules;
