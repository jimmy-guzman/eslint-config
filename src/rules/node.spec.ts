import { nodeRules } from "./node";

test("should create node rules", () => {
  expect(nodeRules).toMatchSnapshot();
});
