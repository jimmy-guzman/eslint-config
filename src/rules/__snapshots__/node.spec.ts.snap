// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`should create node rules 1`] = `
{
  "n/handle-callback-err": [
    "error",
    "^(err|error)$",
  ],
  "n/no-deprecated-api": "error",
  "n/no-exports-assign": "error",
  "n/no-new-require": "error",
  "n/no-path-concat": "error",
  "n/no-process-exit": "off",
  "n/no-top-level-await": [
    "error",
    {
      "ignoreBin": true,
    },
  ],
  "n/prefer-global/console": [
    "error",
    "always",
  ],
  "n/prefer-node-protocol": "error",
  "n/process-exit-as-throw": "error",
}
`;
